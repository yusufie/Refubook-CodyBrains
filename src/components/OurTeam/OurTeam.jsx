import { useEffect, useState } from 'react';
import { Tab } from '@headlessui/react';
import { useTranslation } from 'react-i18next';
// import ourTeam from '../../data/ourTeamData';

const OurTeam = () => {
  const { t } = useTranslation();
  const [selectedTab, setSelectedTab] = useState('All');
  const [countries, setCountries] = useState([]);

  const ourTeam = [
    {
      name: t('Yasin'),
      job: t('Yasin_Job'),
      country: t('Turkey'),
      id: 1,
      link: '',
    },
    {
      name: t('Helin'),
      job: t('Helin_Job'),
      country: t('Turkey'),
      id: 2,
      link: '',
    },
    {
      name: t('Sara'),
      job: t('Sara_Job'),
      country: t('Turkey'),
      id: 3,
      link: '',
    },
    {
      name: t('Yusuf'),
      job: t('Yusuf_Job'),
      country: t('Turkey'),
      id: 4,
      link: '',
    },
    {
      name: t('Asmaa'),
      job: t('Asmaa_Job'),
      country: t('Turkey'),
      id: 5,
      link: '',
    },
    {
      name: t('Salah'),
      job: t('Salah_Job'),
      country: t('Lebanon'),
      id: 6,
      link: '',
    },
    {
      name: t('Kishi'),
      job: t('Kishi_Job'),
      country: t('USA'),
      id: 7,
      link: '',
    },
    {
      name: t('Hiba'),
      job: t('Hiba_Job'),
      country: t('Lebanon'),
      id: 8,
      link: '',
    },
    {
      name: t('Wissam'),
      job: t('Wissam_Job'),
      country: t('USA'),
      id: 9,
      link: '',
    },
  ];

  useEffect(() => {
    // Extract unique countries from the team members data
    const uniqueCountries = Array.from(
      new Set(ourTeam.map((member) => member.country))
    );
    setCountries(uniqueCountries);
  }, []);

  function classNames(...classes) {
    return classes.filter(Boolean).join(' ');
  }
  return (
    <section className="px-8 py-8 bg-refubookAboutBlue text-darkBlue font-bold w-full">
      <div className="container mx-auto">
        <div className="flex  items-center justify-center">
          <h1 className="text-4xl font-bold text-center">{t('Our_Team')}</h1>
        </div>
        <div className="hidden md:block mt-10">
          <Tab.Group>
            <Tab.List className="flex rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full">
              <Tab
                key="All"
                selected={selectedTab === 'All'}
                onClick={() => setSelectedTab('All')}
                className={({ selected }) =>
                  classNames(
                    'w-full rounded-full py-2  text-sm font-medium  ',
                    'ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2',
                    selected
                      ? 'bg-darkBlue  text-refubookAboutBlue shadow'
                      : ' hover:bg-refubookWhite/[0.12] hover:text-refubookBlack'
                  )
                }
              >
                {t('View_All')}
              </Tab>
              {countries.map((country) => (
                <Tab
                  key={country.id}
                  selected={selectedTab === country}
                  onClick={() => setSelectedTab(country)}
                  className={({ selected }) =>
                    classNames(
                      'w-full rounded-full py-2  text-sm font-medium  ',
                      'ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2',
                      selected
                        ? 'bg-darkBlue  text-refubookAboutBlue shadow'
                        : ' hover:bg-refubookWhite/[0.12] hover:text-refubookBlack'
                    )
                  }
                >
                  {country}
                  {/* {t('country')} */}
                </Tab>
              ))}
            </Tab.List>
            <Tab.Panels className="mt-10">
              <Tab.Panel selected={selectedTab === 'All'}>
                <div className="px-4 md:px-48 md:mt-10 md:py-10 justify-self-center items-center w-full mx-auto grid grid-cols-2 gap-16 md:grid-cols-3 md:gap-20">
                  {ourTeam.map((member) => (
                    <div
                      key={member.name}
                      className="flex flex-col items-center justify-center w-full md:ml-8"
                    >
                      <p className="font-medium w-full">{member.name}</p>
                      <p className="font-normal  w-full"> {t(member.job)}</p>
                    </div>
                  ))}
                </div>
              </Tab.Panel>
              {countries.map((country) => (
                <Tab.Panel
                  key={country}
                  selected={selectedTab === country}
                  className="mx-auto"
                >
                  <div className="px-4 md:px-48 md:mt-10 md:py-10 items-center justify-center grid grid-cols-2 gap-16 md:grid-cols-3 md:gap-20">
                    {ourTeam
                      .filter((member) => member.country === country)
                      .map((member) => (
                        <div
                          key={member.name}
                          className="flex flex-col items-center justify-center w-full"
                        >
                          <p className="font-medium  w-full">{member.name}</p>
                          <p className="font-normal  w-full">{member.job}</p>
                        </div>
                      ))}
                  </div>
                </Tab.Panel>
              ))}
            </Tab.Panels>
          </Tab.Group>
        </div>
        <div className="block md:hidden mt-10">
          <Tab.Group>
            <Tab.List className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full">
              <Tab
                key="All"
                selected={selectedTab === 'All'}
                onClick={() => setSelectedTab('All')}
                className={({ selected }) =>
                  classNames(
                    ' w-28 py-2 rounded-full   text-sm font-medium  ',
                    'ring-refubookWhite ring-opacity-60 ring-offset-2 text-refubookWhite  ring-offset-refubookBlue focus:outline-none focus:ring-2',
                    selected
                      ? 'bg-refubookTeamLightRed  shadow'
                      : ' bg-refubookGray hover:text-refubookBlack'
                  )
                }
              >
                {t('View_All')}
              </Tab>
              <Tab.Panel selected={selectedTab === 'All'}>
                <div className="px-4 md:px-48 md:mt-10 md:py-10 justify-self-center items-center w-full mx-auto grid grid-cols-2 gap-16 md:grid-cols-3 md:gap-20">
                  {ourTeam.map((member) => (
                    <div
                      key={member.name}
                      className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
                    >
                      <p className="font-medium w-full">{member.name}</p>
                      <p className="font-normal  w-full">{member.job}</p>
                    </div>
                  ))}
                </div>
              </Tab.Panel>
            </Tab.List>
            <Tab.List className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full">
              {countries.map((country) => (
                <div className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full">
                  <Tab
                    key={country}
                    selected={selectedTab === country}
                    onClick={() => setSelectedTab(country)}
                    className={({ selected }) =>
                      classNames(
                        ' w-28 py-2 rounded-full   text-sm font-medium text-refubookWhite  ',
                        'ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2',
                        selected
                          ? 'bg-refubookTeamLightRed   shadow'
                          : ' bg-refubookGray hover:text-refubookBlack'
                      )
                    }
                  >
                    {country}
                    {/* {t('country')} */}
                  </Tab>

                  <Tab.Panel
                    key={country}
                    selected={selectedTab === country}
                    className="mx-auto"
                  >
                    <div className="px-4 md:px-48 md:mt-10 md:py-10 items-center justify-center grid grid-cols-2 gap-16 md:grid-cols-3 md:gap-20">
                      {ourTeam
                        .filter((member) => member.country === country)
                        .map((member) => (
                          <div
                            key={member.name}
                            className="flex flex-col items-center mt-5 justify-center w-full"
                          >
                            <p className="font-medium  w-full">{member.name}</p>
                            <p className="font-normal  w-full">{member.job}</p>
                          </div>
                        ))}
                    </div>
                  </Tab.Panel>
                </div>
              ))}
            </Tab.List>
          </Tab.Group>
        </div>
      </div>
    </section>
  );
};
export default OurTeam;
