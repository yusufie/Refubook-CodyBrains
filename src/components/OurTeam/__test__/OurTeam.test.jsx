import React from 'react';
import renderer from 'react-test-renderer';
import OurTeam from '../OurTeam';

jest.mock('react-router-dom', () => ({
  Link: 'Link',
}));
describe('OurTeam', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<OurTeam />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
