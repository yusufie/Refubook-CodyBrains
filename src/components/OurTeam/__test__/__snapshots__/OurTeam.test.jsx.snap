// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurTeam renders correctly 1`] = `
<section
  className="px-8 py-8 bg-refubookAboutBlue text-darkBlue font-bold w-full"
>
  <div
    className="container mx-auto"
  >
    <div
      className="flex  items-center justify-center"
    >
      <h1
        className="text-4xl font-bold text-center"
      >
        Our_Team
      </h1>
    </div>
    <div
      className="hidden md:block mt-10"
    >
      <div
        aria-orientation="horizontal"
        className="flex rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full"
        role="tablist"
      >
        <button
          aria-selected={true}
          className="w-full rounded-full py-2  text-sm font-medium   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2 bg-darkBlue  text-refubookAboutBlue shadow"
          data-headlessui-state="selected"
          id="headlessui-tabs-tab-:r0:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="tab"
          selected={true}
          tabIndex={0}
          type="button"
        >
          View_All
        </button>
        <button
          aria-selected={false}
          className="w-full rounded-full py-2  text-sm font-medium   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2  hover:bg-refubookWhite/[0.12] hover:text-refubookBlack"
          data-headlessui-state=""
          id="headlessui-tabs-tab-:r4:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="tab"
          selected={false}
          tabIndex={-1}
          type="button"
        >
          Turkey
        </button>
        <button
          aria-selected={false}
          className="w-full rounded-full py-2  text-sm font-medium   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2  hover:bg-refubookWhite/[0.12] hover:text-refubookBlack"
          data-headlessui-state=""
          id="headlessui-tabs-tab-:r5:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="tab"
          selected={false}
          tabIndex={-1}
          type="button"
        >
          Lebanon
        </button>
        <button
          aria-selected={false}
          className="w-full rounded-full py-2  text-sm font-medium   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2  hover:bg-refubookWhite/[0.12] hover:text-refubookBlack"
          data-headlessui-state=""
          id="headlessui-tabs-tab-:r6:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="tab"
          selected={false}
          tabIndex={-1}
          type="button"
        >
          USA
        </button>
      </div>
      <div
        className="mt-10"
      >
        <div
          data-headlessui-state="selected"
          id="headlessui-tabs-panel-:r1:"
          role="tabpanel"
          selected={true}
          tabIndex={0}
        >
          <div
            className="px-4 md:px-48 md:mt-10 md:py-10 justify-self-center items-center w-full mx-auto grid grid-cols-2 gap-16 md:grid-cols-3 md:gap-20"
          >
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Yasin
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Yasin_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Helin
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Helin_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Sara
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Sara_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Yusuf
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Yusuf_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Asmaa
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Asmaa_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Salah
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Salah_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Kishi
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Kishi_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Hiba
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Hiba_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Wissam
              </p>
              <p
                className="font-normal  w-full"
              >
                 
                Wissam_Job
              </p>
            </div>
          </div>
        </div>
        <span
          id="headlessui-tabs-panel-:r7:"
          role="tabpanel"
          style={
            Object {
              "borderWidth": "0",
              "clip": "rect(0, 0, 0, 0)",
              "height": 0,
              "left": 1,
              "margin": -1,
              "overflow": "hidden",
              "padding": 0,
              "position": "fixed",
              "top": 1,
              "whiteSpace": "nowrap",
              "width": 1,
            }
          }
          tabIndex={-1}
        />
        <span
          id="headlessui-tabs-panel-:r8:"
          role="tabpanel"
          style={
            Object {
              "borderWidth": "0",
              "clip": "rect(0, 0, 0, 0)",
              "height": 0,
              "left": 1,
              "margin": -1,
              "overflow": "hidden",
              "padding": 0,
              "position": "fixed",
              "top": 1,
              "whiteSpace": "nowrap",
              "width": 1,
            }
          }
          tabIndex={-1}
        />
        <span
          id="headlessui-tabs-panel-:r9:"
          role="tabpanel"
          style={
            Object {
              "borderWidth": "0",
              "clip": "rect(0, 0, 0, 0)",
              "height": 0,
              "left": 1,
              "margin": -1,
              "overflow": "hidden",
              "padding": 0,
              "position": "fixed",
              "top": 1,
              "whiteSpace": "nowrap",
              "width": 1,
            }
          }
          tabIndex={-1}
        />
      </div>
    </div>
    <div
      className="block md:hidden mt-10"
    >
      <div
        aria-orientation="horizontal"
        className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full"
        role="tablist"
      >
        <button
          aria-selected={true}
          className=" w-28 py-2 rounded-full   text-sm font-medium   ring-refubookWhite ring-opacity-60 ring-offset-2 text-refubookWhite  ring-offset-refubookBlue focus:outline-none focus:ring-2 bg-refubookTeamLightRed  shadow"
          data-headlessui-state="selected"
          id="headlessui-tabs-tab-:r2:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="tab"
          selected={true}
          tabIndex={0}
          type="button"
        >
          View_All
        </button>
        <div
          data-headlessui-state="selected"
          id="headlessui-tabs-panel-:r3:"
          role="tabpanel"
          selected={true}
          tabIndex={0}
        >
          <div
            className="px-4 md:px-48 md:mt-10 md:py-10 justify-self-center items-center w-full mx-auto grid grid-cols-2 gap-16 md:grid-cols-3 md:gap-20"
          >
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Yasin
              </p>
              <p
                className="font-normal  w-full"
              >
                Yasin_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Helin
              </p>
              <p
                className="font-normal  w-full"
              >
                Helin_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Sara
              </p>
              <p
                className="font-normal  w-full"
              >
                Sara_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Yusuf
              </p>
              <p
                className="font-normal  w-full"
              >
                Yusuf_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Asmaa
              </p>
              <p
                className="font-normal  w-full"
              >
                Asmaa_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Salah
              </p>
              <p
                className="font-normal  w-full"
              >
                Salah_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Kishi
              </p>
              <p
                className="font-normal  w-full"
              >
                Kishi_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Hiba
              </p>
              <p
                className="font-normal  w-full"
              >
                Hiba_Job
              </p>
            </div>
            <div
              className="flex flex-col items-center justify-center mt-5 w-full md:ml-8"
            >
              <p
                className="font-medium w-full"
              >
                Wissam
              </p>
              <p
                className="font-normal  w-full"
              >
                Wissam_Job
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-orientation="horizontal"
        className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full"
        role="tablist"
      >
        <div
          className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full"
        >
          <button
            aria-selected={false}
            className=" w-28 py-2 rounded-full   text-sm font-medium text-refubookWhite   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2  bg-refubookGray hover:text-refubookBlack"
            data-headlessui-state=""
            id="headlessui-tabs-tab-:ra:"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="tab"
            selected={false}
            tabIndex={-1}
            type="button"
          >
            Turkey
          </button>
          <span
            id="headlessui-tabs-panel-:rb:"
            role="tabpanel"
            style={
              Object {
                "borderWidth": "0",
                "clip": "rect(0, 0, 0, 0)",
                "height": 0,
                "left": 1,
                "margin": -1,
                "overflow": "hidden",
                "padding": 0,
                "position": "fixed",
                "top": 1,
                "whiteSpace": "nowrap",
                "width": 1,
              }
            }
            tabIndex={-1}
          />
        </div>
        <div
          className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full"
        >
          <button
            aria-selected={false}
            className=" w-28 py-2 rounded-full   text-sm font-medium text-refubookWhite   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2  bg-refubookGray hover:text-refubookBlack"
            data-headlessui-state=""
            id="headlessui-tabs-tab-:rc:"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="tab"
            selected={false}
            tabIndex={-1}
            type="button"
          >
            Lebanon
          </button>
          <span
            id="headlessui-tabs-panel-:rd:"
            role="tabpanel"
            style={
              Object {
                "borderWidth": "0",
                "clip": "rect(0, 0, 0, 0)",
                "height": 0,
                "left": 1,
                "margin": -1,
                "overflow": "hidden",
                "padding": 0,
                "position": "fixed",
                "top": 1,
                "whiteSpace": "nowrap",
                "width": 1,
              }
            }
            tabIndex={-1}
          />
        </div>
        <div
          className="flex flex-col rounded-xl p-1 max-w-lg mx-auto mt-5 justify-center items-center w-full"
        >
          <button
            aria-selected={false}
            className=" w-28 py-2 rounded-full   text-sm font-medium text-refubookWhite   ring-refubookWhite ring-opacity-60 ring-offset-2 ring-offset-refubookBlue focus:outline-none focus:ring-2  bg-refubookGray hover:text-refubookBlack"
            data-headlessui-state=""
            id="headlessui-tabs-tab-:re:"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="tab"
            selected={false}
            tabIndex={-1}
            type="button"
          >
            USA
          </button>
          <span
            id="headlessui-tabs-panel-:rf:"
            role="tabpanel"
            style={
              Object {
                "borderWidth": "0",
                "clip": "rect(0, 0, 0, 0)",
                "height": 0,
                "left": 1,
                "margin": -1,
                "overflow": "hidden",
                "padding": 0,
                "position": "fixed",
                "top": 1,
                "whiteSpace": "nowrap",
                "width": 1,
              }
            }
            tabIndex={-1}
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
